pkg/controller/portworxdiag/controller_test.go

```diff
@@ -25,13 +38,199 @@ import (
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/record"
 	k8scontroller "k8s.io/kubernetes/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/libopenstorage/operator/pkg/client/clientset/versioned/scheme"
 	testutil "github.com/libopenstorage/operator/pkg/util/test"
 )
 
+const (
+	etcHostsFile       = "/etc/hosts"
+	tempEtcHostsMarker = "### px-operator unit-test"
+)
+
+var (
+	podCreateCount = 0
+)
+
+// setupEtcHosts sets up given ip/hosts in `/etc/hosts` file for "local" DNS resolution  (i.e. emulate K8s DNS)
```

Comment:
Is there any way to avoid this in the tests? Are we trying to emulate k8s DNS because the production code references the PX Service? 

If so, maybe we can make the connection information configurable instead and point to localhost:9021 for testing?

Reply:
There may be a way to avoid it, but this is also what the StorageCluster controller tests do, so it's already an evil we have :P

Reply:
Ah ok, can we externalize this here then? 
https://github.com/libopenstorage/operator/blob/master/pkg/util/test/util.go

At least in this case, fixes to any issues here would go in one spot.